cohorts/functions.py

```diff
@@ -68,150 +68,128 @@ def get_patient_to_mb(cohort):
-        return (isinstance(filterable_effect.effect, Exonic) and
-                filterable_effect.variant.is_deletion and
-                filter_fn(filterable_effect, **kwargs))
-    # This only loads one effect per variant.
-    patient_id = row["patient_id"]
-    return cohort.load_effects(
-        only_nonsynonymous=True,
-        patients=[cohort.patient_from_id(patient_id)],
-        filter_fn=exonic_deletion_filter_fn,
-        **kwargs)
-
-@count_function
-def exonic_insertion_count(row, cohort, filter_fn, normalized_per_mb, **kwargs):
-    def exonic_insertion_filter_fn(filterable_effect, **kwargs):
-        assert filter_fn is not None, "filter_fn should never be None, but it is."
-        return (isinstance(filterable_effect.effect, Exonic) and
-                filterable_effect.variant.is_insertion and
-                filter_fn(filterable_effect, **kwargs))
-    # This only loads one effect per variant.
-    patient_id = row["patient_id"]
-    return cohort.load_effects(
-        only_nonsynonymous=True,
-        patients=[cohort.patient_from_id(patient_id)],
-        filter_fn=exonic_insertion_filter_fn,
-        **kwargs)
+def count_variants_function_builder(function_name, filterable_variant_function=None):
+    @count_function
+    def count(row, cohort, filter_fn, normalized_per_mb, **kwargs):
+        def count_filter_fn(filterable_variant, **kwargs):
+            assert filter_fn is not None, "filter_fn should never be None, but it is."
+            return ((filterable_variant_function(filterable_variant) if filterable_variant_function is not None else True) and
+                    filter_fn(filterable_variant, **kwargs))
+        patient_id = row["patient_id"]
+        return cohort.load_variants(
+            patients=[cohort.patient_from_id(patient_id)],
+            filter_fn=count_filter_fn,
+            **kwargs)
+    count.__name__ = function_name
+    return count
+
+def count_effects_function_builder(function_name, only_nonsynonymous, filterable_effect_function=None):
+    @count_function
+    def count(row, cohort, filter_fn, normalized_per_mb, **kwargs):
+        def count_filter_fn(filterable_effect, **kwargs):
+            assert filter_fn is not None, "filter_fn should never be None, but it is."
+            return ((filterable_effect_function(filterable_effect) if filterable_effect_function is not None else True) and
+                    filter_fn(filterable_effect, **kwargs))
+        # This only loads one effect per variant.
+        patient_id = row["patient_id"]
+        return cohort.load_effects(
+            only_nonsynonymous=only_nonsynonymous,
+            patients=[cohort.patient_from_id(patient_id)],
+            filter_fn=count_filter_fn,
+            **kwargs)
+    count.__name__ = function_name
+    return count
+
+variant_count = count_variants_function_builder("variant_count")
+
+snv_count = count_variants_function_builder(
+    "snv_count",
```

Comment:
it would be nice to include an optional docstring for these generated functions. Perhaps you can even construct a generic string based on contents of `filterable_variant_function` ?